### ISSUE TITLE
OpenLens 6.3.0 - No Logs or Shell buttons

### ISSUE THREAD

**Issue by lukjaw** [kontena/lens#6823]
**Describe the bug**
After installing OpenLens v 6.3.0 i cannot see Logs and Shell buttons under pods / nodes

**To Reproduce**
1. Install OpenLens on Windows using scoop
2. Launching OpenLens

**Expected behavior**
I would like to see the Logs and Shell buttons like in standard Lens IDE

**Screenshots**
![image](https://user-images.githubusercontent.com/18211802/209138967-9d571fa0-731a-4bce-8048-011d5673837a.png)

**Environment (please complete the following information):**
- Lens Version: OpenLens v 6.3.0
- OS: Windows 10 Build 19044.2364
- Installation method (e.g. snap or AppImage in Linux): scoop

**Logs:**
When you run the application executable from command line you will see some logging output. Please paste them here:
```
info:    Ôľ¬ ­čöî Starting LensProxy +0ms
info:    Ôľ¬ [LENS-PROXY]: Proxy server has started at 127.0.0.1:53779 +5ms
info:    Ôľ¬ ­čöÄ Testing LensProxy connection ... +2ms
info:    Ôľ¬ [LENS-USER-STORE]: LOADING ... +7ms
info:    Ôľ¬ [LENS-USER-STORE]: LOADED from C:\Users\lukjaw\AppData\Roaming\OpenLens\lens-user-store.json +14ms
info:    Ôľ¬ [LENS-CLUSTER-STORE]: LOADING ... +0ms
info:    Ôľ¬ [LENS-CLUSTER-STORE]: LOADED from C:\Users\lukjaw\AppData\Roaming\OpenLens\lens-cluster-store.json +140ms
info:    Ôľ¬ ÔÜí LensProxy connection OK +16ms
info:    Ôľ¬ [CREATE-ELECTRON-WINDOW]: Loading content for window "splash" from file: C:\Users\lukjaw\scoop\apps\openlens\current\resources\app.asar\static\splash.html... +48ms
info:    Ôľ¬ [CREATE-ELECTRON-WINDOW]: Window "splash" loaded +222ms
info:    Ôľ¬ [LENS-EXTENSIONS]: LOADING ... +21ms
info:    Ôľ¬ [LENS-EXTENSIONS]: LOADED from C:\Users\lukjaw\AppData\Roaming\OpenLens\lens-extensions.json +3ms
info:    Ôľ¬ [LENS-WEBLINK-STORE]: LOADING ... +5ms
info:    Ôľ¬ [LENS-WEBLINK-STORE]: LOADED from C:\Users\lukjaw\AppData\Roaming\OpenLens\lens-weblink-store.json +12ms
info:    Ôľ¬ ­čôč Setting protocol client for lens:// +8ms
info:    Ôľ¬ ­čôč Protocol client register succeeded Ôťů +2ms
info:    Ôľ¬ ­čžę Initializing extensions +4ms
info:    Ôľ¬ [LENS-FILESYSTEM-PROVISIONER-STORE]: LOADING ... +41ms
info:    Ôľ¬ [LENS-FILESYSTEM-PROVISIONER-STORE]: LOADED from C:\Users\lukjaw\AppData\Roaming\OpenLens\lens-filesystem-provisioner-store.json +4ms
info:    Ôľ¬ ­čÉÜ Syncing shell environment +0ms
info:    Ôľ¬ [EXTENSIONS-LOADER]: auto initializing extensions +23ms
info:    Ôľ¬ [LENS-HOTBAR-STORE]: LOADING ... +2ms
info:    Ôľ¬ [LENS-HOTBAR-STORE]: LOADED from C:\Users\lukjaw\AppData\Roaming\OpenLens\lens-hotbar-store.json +10ms
info:    Ôľ¬ [EXTENSION-DISCOVERY] loading extensions from C:\Users\lukjaw\AppData\Roaming\OpenLens +9ms
info:    Ôľ¬ [EXTENSION-DISCOVERY] watching extension add/remove in C:\Users\lukjaw\.k8slens\extensions +12ms
info:    Ôľ¬ [CREATE-ELECTRON-WINDOW]: Loading content for window "first-application-window" from url: http://localhost:53779... +17ms
info:    Ôľ¬ [CREATE-ELECTRON-WINDOW]: Window "first-application-window" loaded +2s
Checking for update
info:    Ôľ¬ [KUBECONFIG-SYNC]: starting requested syncs +444ms
info:    ÔöĆ [KUBECONFIG-SYNC]: starting sync of file/folder +2ms
info:    ÔöŚ [1] { filePath: 'C:\\Users\\lukjaw\\AppData\\Roaming\\OpenLens\\kubeconfigs' }
info:    ÔöĆ [KUBECONFIG-SYNC]: starting sync of file/folder +2ms
info:    ÔöŚ [1] { filePath: 'C:\\Users\\lukjaw\\.kube' }

 
Error: Error: Cannot find channel "latest.yml" update info: HttpError: 404
"method: GET url: https://github.com/MuhammedKalkan/OpenLens/releases/download/Latest/latest.yml?noCache=1gkstml0t\n\nPlease double check that your authentication token is correct. Due to security reasons, actual status maybe not reported, but 404.\n"
Headers: {
  "server": "GitHub.com",
  "date": "Thu, 22 Dec 2022 12:42:34 GMT",
  "content-type": "text/plain; charset=utf-8",
  "vary": "X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame, Accept-Encoding, Accept, X-Requested-With",
  "cache-control": "no-cache",
  "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
  "x-frame-options": "deny",
  "x-content-type-options": "nosniff",
  "x-xss-protection": "0",
  "referrer-policy": "no-referrer-when-downgrade",
  "content-security-policy": "default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'",
  "content-encoding": "gzip",
  "content-length": "29",
  "x-github-request-id": "D228:79F3:8527612:893D291:63A450BA"
}
    at createHttpError (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\httpExecutor.js:14:12)
    at ElectronHttpExecutor.handleResponse (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\httpExecutor.js:116:20)
    at ClientRequest.<anonymous> (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\httpExecutor.js:82:26)
    at ClientRequest.emit (node:events:526:28)
    at ClientRequest.emit (node:domain:475:12)
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:101:6917)
    at SimpleURLLoaderWrapper.emit (node:events:526:28)
    at SimpleURLLoaderWrapper.emit (node:domain:475:12)
    at Object.newError (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\index.js:47:19)
    at GenericProvider.getLatestVersion (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\out\providers\GenericProvider.js:27:50)
    at async NsisUpdater.getUpdateInfoAndProvider (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\out\AppUpdater.js:298:19)
    at async NsisUpdater.doCheckForUpdates (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\out\AppUpdater.js:312:24)
    at async C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\static\build\main.js:2:413995
    at async r (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\static\build\main.js:2:414485)
    at async C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\static\build\main.js:2:417149
error:   ÔöĆ [UPDATE-APP/CHECK-FOR-UPDATES] Cannot find channel "latest.yml" update info: HttpError: 404
"method: GET url: https://github.com/MuhammedKalkan/OpenLens/releases/download/Latest/latest.yml?noCache=1gkstml0t\n\nPlease double check that your authentication token is correct. Due to security reasons, actual status maybe not reported, but 404.\n"
Headers: {
  "server": "GitHub.com",
  "date": "Thu, 22 Dec 2022 12:42:34 GMT",
  "content-type": "text/plain; charset=utf-8",
  "vary": "X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame, Accept-Encoding, Accept, X-Requested-With",
  "cache-control": "no-cache",
  "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
  "x-frame-options": "deny",
  "x-content-type-options": "nosniff",
  "x-xss-protection": "0",
  "referrer-policy": "no-referrer-when-downgrade",
  "content-security-policy": "default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'",
  "content-encoding": "gzip",
  "content-length": "29",
  "x-github-request-id": "D228:79F3:8527612:893D291:63A450BA"
}
    at createHttpError (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\httpExecutor.js:14:12)
    at ElectronHttpExecutor.handleResponse (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\httpExecutor.js:116:20)
    at ClientRequest.<anonymous> (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\httpExecutor.js:82:26)
    at ClientRequest.emit (node:events:526:28)
    at ClientRequest.emit (node:domain:475:12)
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:101:6917)
    at SimpleURLLoaderWrapper.emit (node:events:526:28)
    at SimpleURLLoaderWrapper.emit (node:domain:475:12) +286ms
error:   Ôöâ [ 1] Error: Cannot find channel "latest.yml" update info: HttpError: 404
error:   Ôöâ [ 2] "method: GET url: https://github.com/MuhammedKalkan/OpenLens/releases/download/Latest/latest.yml?noCache=1gkstml0t\n\nPlease double check that your authentication token is correct. Due to security reasons, actual status maybe not reported, but 404.\n"
error:   Ôöâ [ 3] Headers: {
error:   Ôöâ [ 4]   "server": "GitHub.com",
error:   Ôöâ [ 5]   "date": "Thu, 22 Dec 2022 12:42:34 GMT",
error:   Ôöâ [ 6]   "content-type": "text/plain; charset=utf-8",
error:   Ôöâ [ 7]   "vary": "X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame, Accept-Encoding, Accept, X-Requested-With",
error:   Ôöâ [ 8]   "cache-control": "no-cache",
error:   Ôöâ [ 9]   "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
error:   Ôöâ [10]   "x-frame-options": "deny",
error:   Ôöâ [11]   "x-content-type-options": "nosniff",
error:   Ôöâ [12]   "x-xss-protection": "0",
error:   Ôöâ [13]   "referrer-policy": "no-referrer-when-downgrade",
error:   Ôöâ [14]   "content-security-policy": "default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'",
error:   Ôöâ [15]   "content-encoding": "gzip",
error:   Ôöâ [16]   "content-length": "29",
error:   Ôöâ [17]   "x-github-request-id": "D228:79F3:8527612:893D291:63A450BA"
error:   Ôöâ [18] }
error:   Ôöâ [19]     at createHttpError (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\httpExecutor.js:14:12)
error:   Ôöâ [20]     at ElectronHttpExecutor.handleResponse (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\httpExecutor.js:116:20)
error:   Ôöâ [21]     at ClientRequest.<anonymous> (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\httpExecutor.js:82:26)
error:   Ôöâ [22]     at ClientRequest.emit (node:events:526:28)
error:   Ôöâ [23]     at ClientRequest.emit (node:domain:475:12)
error:   Ôöâ [24]     at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:101:6917)
error:   Ôöâ [25]     at SimpleURLLoaderWrapper.emit (node:events:526:28)
error:   Ôöâ [26]     at SimpleURLLoaderWrapper.emit (node:domain:475:12)
error:   Ôöâ [27]     at Object.newError (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\out\index.js:47:19)
error:   Ôöâ [28]     at GenericProvider.getLatestVersion (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\out\providers\GenericProvider.js:27:50)
error:   Ôöâ [29]     at async NsisUpdater.getUpdateInfoAndProvider (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\out\AppUpdater.js:298:19)
error:   Ôöâ [30]     at async NsisUpdater.doCheckForUpdates (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\node_modules\electron-updater\out\AppUpdater.js:312:24)
error:   Ôöâ [31]     at async C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\static\build\main.js:2:413995
error:   Ôöâ [32]     at async r (C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\static\build\main.js:2:414485)
error:   Ôöâ [33]     at async C:\Users\lukjaw\scoop\apps\openlens\6.3.0\resources\app.asar\static\build\main.js:2:417149
error:   ÔöŚ [34] { code: 'ERR_UPDATER_CHANNEL_FILE_NOT_FOUND' }
```

**Additional context**
I Was using Lens IDE on a license grace period and i wanted to switch to OpenLens. 

I removed previous Lens version and cleared all /User/Appdata files and any remaining files from old installations, rebooted, re-installed OpenLens again, tried also solution mentioned in here: 

https://github.com/lensapp/lens/issues/1690#issuecomment-886260245

But i ran `npm install` from the scoop folder (C:\Users\****\scoop\apps\openlens\6.3.0\resources\app.asar.unpacked\node_modules\)

**Comment by dedo1911** [comment]
Same issue with clean setup using OpenLens-6.3.0.exe
I wasn't able to test on other platforms yet.

**Comment by odedonato** [comment]
> Same issue with clean setup using OpenLens-6.3.0.exe I wasn't able to test on other platforms yet.

On MacOS same thing.

@dedo1911 I may know you from some Ingress events, I'm Dedo. :)

**Comment by CappyT** [comment]
Same issue here on windows, using installer.

**Comment by mossroy** [comment]
Same on Ubuntu. It looks like a regression of version 6.3.0

**Comment by msa0311** [comment]
The in-tree extensions have been removed from with 6.3.0, see the following issues / PRs that explain the background:

- https://github.com/lensapp/lens/issues/6749
- https://github.com/lensapp/lens/pull/6715
- https://github.com/lensapp/lens/pull/6775

Functionality is still available with the latest stable binary version from our website https://k8slens.dev/.

**Comment by jkroepke** [comment]
@msa0311 can you explain the reason for this?

**Comment by holograph** [comment]
> The in-tree extensions have been removed from with 6.3.0, see the following issues / PRs that explain the background:
> 
>     * [Improve Extension loading capabilities #6749](https://github.com/lensapp/lens/issues/6749)
> 
>     * [Include bundled extensions in package.json #6715](https://github.com/lensapp/lens/pull/6715)
> 
>     * [Remove in-tree extensions to help facilitate a more secure and faster booting lens #6775](https://github.com/lensapp/lens/pull/6775)
> 
> 
> Functionality is still available with the latest stable binary version from our website https://k8slens.dev/.

That's complete bullshit. You took an open-source project and made it un-buildable for commercial purposes. Own it.

**Comment by malkir** [comment]
> The in-tree extensions have been removed from with 6.3.0, see the following issues / PRs that explain the background
> Functionality is still available with the latest stable binary version from our website

..and just like that, my evaluation of the product is complete, straight to /dev/null with Lens, at least with a dev team trying to cover their crap so sloppily.

**Comment by worldofgeese** [comment]
For those of you looking for an alternative, I'll be looking at Monokle Desktop, which just released "[Cluster Mode](https://monokle.io/blog/monokle-1-13-release)". I haven't tried it yet but thought I'd share. Are there any other GUI clients?

**Comment by sherifabdlnaby** [comment]
Seriously folks ? That's exactly how to lose your community (and potential customers)! That's not how you do Open Source.

**Comment by aldmbmtl** [comment]
My team and I, a pretty large global team, have been evaluating Lens for the last 3 months and it was looking really promising. This change has made our decision for us. We will look elsewhere now.

**Comment by sigmonsays** [comment]
anyone gonna fork this tool now?

**Comment by aldmbmtl** [comment]
There is a fork called OpenLens. I switched to that awhile ago when Lens started requiring a login just to use it. I have been using Lens for about a year and a half at home (and loved it). That is why I recommended it to my company (an example of why the community is important....). But with this happening, I can't even use it personally. I can't check my logs or shell in without needing to do a rebuild or digging into the source code? So why would I stick my neck out and recommend it to my employer? 

I am personally switching back to k9s for now and trying out Monokle. I may come back to Lens if they fix this, but I surely will not be recommending it anymore in a professional setting. We are starting to have conversations about alternatives since Lens is no longer an option for us.

**Comment by jakolehm** [comment]
> That's complete bullshit. You took an open-source project and made it un-buildable for commercial purposes. Own it.

The open-source project (this repo) is still buildable, we even provide a way to build it with a single command.

**Comment by holograph** [comment]
> > That's complete bullshit. You took an open-source project and made it un-buildable for commercial purposes. Own it.
> 
> The open-source project (this repo) is still buildable, we even provide a way to build it with a single command.

... and require out-of-band account management with the prebuilt binaries, while making it harder for open source maintainers like @MuhammedKalkan to keep binary builds going. I appreciate Lens and the work put into it a lot, which is exactly why these latest moves feel like a big middle finger to the existing community.

**Comment by jkroepke** [comment]
> > That's complete bullshit. You took an open-source project and made it un-buildable for commercial purposes. Own it.
> 
> The open-source project (this repo) is still buildable, we even provide a way to build it with a single command.

And why you remove existing code? I read the reason the PR for technical reasons. But then in the same post, you said its available in the Lens Desktop binary builds. Can you explain @jakolehm why the technical reasons are not effected by Lens Desktop? You removed the extensions to save startup time, but re-introduce them downstream? Why?

**Comment by jakolehm** [comment]
> And why you remove existing code?

I don't want to sound too snarky but code is being removed (and added) every day, it's the normal practice with software development.  If someone needs the removed code it is available via version control (git).

> I read the reason the PR for technical reasons. But then in the same post, you said its available in the Lens Desktop binary builds. Can you explain @jakolehm why the technical reasons are not effected by Lens Desktop?

Lens Desktop includes many extensions that are not part of this repo (even before 6.3.0). Technical reasons for removing extensions from this repo should be pretty clear (as described in the linked issues/PRs) and everyone can still install those extensions manually to OpenLens if they desire to do so.

> You removed the extensions to save startup time, but re-introduce them downstream? Why?

Startup time improvements have nothing to do directly with number of extensions being included in a build, we actually improved the performance and stability of bundled extensions loader.

**Comment by AssafKatz3** [comment]
> The in-tree extensions have been removed from with 6.3.0, see the following issues / PRs that explain the background:
> 
> * [Improve Extension loading capabilities #6749](https://github.com/lensapp/lens/issues/6749)
> * [Include bundled extensions in package.json #6715](https://github.com/lensapp/lens/pull/6715)
> * [Remove in-tree extensions to help facilitate a more secure and faster booting lens #6775](https://github.com/lensapp/lens/pull/6775)
> 
> Functionality is still available with the latest stable binary version from our website https://k8slens.dev/.

@msa0311 These reasons explained why you removed them from the installer, not why these extensions aren't available for installing.

**Comment by jkroepke** [comment]
@jakolehm 

> everyone can still install those extensions manually to OpenLens if they desire to do so.

Then please provide the actual source code location of the extensions that are used in Lens Desktop.

**Comment by kdeyko** [comment]
@jakolehm 

> everyone can still install those extensions manually to OpenLens

How do I do it? Could you please add the instruction to Readme or anywhere else? I believe such major changes to functionality should be reflected in the documentation.

**Comment by AssafKatz3** [comment]
@kdeyko This should be in the extensions list, I opened a case for it (linked above).

**Comment by Carus11** [comment]
~~I understand that this isn't my project and that Lens is within their rights to monetize the project. Neutering the functionality of the open source version is disappointing though. I filled out the surveys on what features I would pay for but I never imagined that basic functionality would be removed instead.~~

~~This will likely lead to me not using lens at all, and not being in a position to recommend it to my clients as the tool I once loved to use.~~ 

I was mistaken in my original understanding of the situation, better conversation below.

**Comment by EvertonSA** [comment]
nice, I spent so much time advocating for lens, now this. thanks. will drop and discourage the usage on my current employer

**Comment by marafa-sugarcrm** [comment]
just yesterday i added lens to our internal wiki of apps to use. and just now i removed it

**Comment by cambierr** [comment]
Are you guys for real ?

We were evaluating subscribing to lens for all our support & infrastructure teams... This wild move really makes me think again about investing money in a software driven in such a way...

**Comment by jakolehm** [comment]
> @jakolehm
> 
> > everyone can still install those extensions manually to OpenLens
> 
> How do I do it? Could you please add the instruction to Readme or anywhere else? I believe such major changes to functionality should be reflected in the documentation.

See https://github.com/alebcay/openlens-extensions .

**Comment by AssafKatz3** [comment]
> > @jakolehm
> > > everyone can still install those extensions manually to OpenLens
> > 
> > 
> > How do I do it? Could you please add the instruction to Readme or anywhere else? I believe such major changes to functionality should be reflected in the documentation.
> 
> See https://github.com/alebcay/openlens-extensions .

@jakolehm 
I did it, I had some problems with Windows, but the real issue is that it is very complex to install. This should be similar to other extensions, I opened a case for it (see above).

**Comment by dkulchinsky** [comment]
> > @jakolehm
> > > everyone can still install those extensions manually to OpenLens
> > 
> > 
> > How do I do it? Could you please add the instruction to Readme or anywhere else? I believe such major changes to functionality should be reflected in the documentation.
> 
> See [alebcay/openlens-extensions](https://github.com/alebcay/openlens-extensions) .

@jakolehm providing this as a reference/source for the removed extensions is very misleading and frankly disingenuous. The repo you linked is one person's attempt (source: https://github.com/MuhammedKalkan/OpenLens/issues/83#issuecomment-1366422225 & https://github.com/MuhammedKalkan/OpenLens/issues/83#issuecomment-1366876015) to try and restore what you folks removed from the Lens Core repo, but didn't provide any instructions or relevant code on how users of the Core project (or derivative such as OpenLens) can utilize.

If these extensions are no longer part of the core project, they should have a proper home (probably in https://github.com/lensapp/lens-extensions?) where users can install them without complex build instructions and more importantly without relying on someone else's efforts.

/cc @alebcay

**Comment by jakolehm** [comment]
> @jakolehm providing this as a reference/source for the removed extensions is very misleading and frankly disingenuous. The repo you linked is one person's attempt (source: https://github.com/MuhammedKalkan/OpenLens/issues/83#issuecomment-1366422225 & https://github.com/MuhammedKalkan/OpenLens/issues/83#issuecomment-1366876015) to try and restore what you folks removed from the Lens Core repo, but didn't provide any instructions or relevant code on how users of the Core project (or derivative such as OpenLens) can utilize.

@dkulchinsky I was just trying to help folks who are trying find a way to install those extensions, I'm sorry if I was giving a misleading information there.

> If these extensions are no longer part of the core project, they should have a proper home (probably in https://github.com/lensapp/lens-extensions?) where users can install them without complex build instructions and more importantly without relying on someone else's efforts.

I believe nothing stops for doing that. The only issue is who would be willing to maintain these? Unless we can identify maintainers, it might be better to make them available outside of lensapp organization. Or have them under lensapp org but indicate we need maintainers.

**Comment by kdeyko** [comment]
@jakolehm

>The only issue is who would be willing to maintain these?

But the mentioned plugins are included in the Lens Desktop, aren't they?
If yes, it means that they **are already** maintained by you guys?
And if yes, it means that you effectively close-sourced them? 🤔

**Comment by dkulchinsky** [comment]
> I believe nothing stops for doing that. The only issue is who would be willing to maintain these? Unless we can identify maintainers, it might be better to make them available outside of lensapp organization. Or have them under lensapp org but indicate we need maintainers.

@jakolehm sorry, but this is a very circular conversation we're all having here.

From where I'm sitting, the summary of the situation is:
1. the extensions being discussed here were removed from the Lens Core repo
2. they still exist, but are provided only by the proprietary Lens Desktop app
3. your organization would not make these extensions or their source code available (except for what's already in GitHub in the removed code, but that is now becoming stale and would be difficult to maintain) for others to install if they use a Lens Core based app (like OpenLens)

in your comment [here](https://github.com/lensapp/lens/issues/6823#issuecomment-1366580127) you said:
> everyone can still install those extensions manually to OpenLens if they desire to do so.

the above is obviously not the case, as these extensions are not available for installation and we don't have their (updated) source code, except what can be retrieved from the now removed code which is already stale, so this would require someone to reverse-engineer them and maintain their compatibility going forward.

> it might be better to make them available outside of lensapp organization. Or have them under lensapp org but indicate we need maintainers.

This comment also confuses me quite a bit, you already maintain them 😄 since as you point out, they are available in Lens Desktop. You just have no intention to provide them for non Desktop Lens users.

The bottom line is that these functionalities, which many here argue are very basic/core to this tool (and I tend to agree), were removed from Lens Core (thus no longer available on OpenLens) and users of the Open Source variant of Lens that want these functionalities back would have to develop and maintain them on their own.

Personally, I think it's a shame, but you have the choice/right to do this.

Looking at the comments here and all the other related issues, you clearly didn't handle this properly and might be worth pondering on that a bit, but then again, I assume you don't really focus/target any of us here using the Open Source variants of your app, so I'm guessing this is not really going to affect you much.

**Comment by jkroepke** [comment]
@jakolehm if you want to share the real reason, e.g. The extensions are closed source now, sources are not longer availble for bussiness reasons. Everyone will understand this and may stop asking for sources. 

the whole Community here does not understand the reason, why extensions are removed for technical reasons while there exists does streaam. 


@jakolehm are you accpeting PR to re integrate the extensions?

**Comment by treyhendon** [comment]
This is also a shame because the licensed version of Lens is much heaver on system resources than OpenLens that I built with the Makefile in this repo.  I assume it's because of the extra stuff like Workspaces in the licensed version, but whatever the reason, I prefer OpenLens for it's performance benefits.

**Comment by marafa-sugarcrm** [comment]
> Looking at the comments here and all the other related issues, you clearly didn't handle this properly and might be worth pondering on that a bit, but then again, I assume you don't really focus/target any of us here using the Open Source variants of your app, so I'm guessing this is not really going to affect you much.

no. and its not the first time which is why a lot of ppl are now using [openlens](https://github.com/MuhammedKalkan/OpenLens)

**Comment by marafa-sugarcrm** [comment]
IANAL so i am asking. is it legal for mirantis to remove _some_ opensource code and change it to commercial? i thought it was an all or none situation

**Comment by alebcay** [comment]
> I believe nothing stops for doing that. The only issue is who would be willing to maintain these? Unless we can identify maintainers, it might be better to make them available outside of lensapp organization. Or have them under lensapp org but indicate we need maintainers.

Users of OpenLens or other Lens Core-based applications can add extension `@alebcay/openlens-node-pod-menu` to get this feature back for now.

I don't understand how maintainership is an issue here. Someone is clearly already maintaining this extension/feature for the proprietary Lens Desktop (presumably the same person(s) that maintained it prior to 6.3.0). This change also requires deliberate extra steps for the folks at Mirantis to _put back_ the node/pod menu extensions in the Desktop version - this does not appear to be some case of "oops we moved these crucial features out of the core repo into separate source repos and forgot to publish the new source repos".

Frankly, I'm not comfortable with the thought of these extensions coming under the lensapp org. From the actions so far, it seems like Mirantis intends to make these features only available in their proprietary Desktop version. With that intention in mind, having an open source version of these extensions in the lensapp org poses a conflict of interest. I would be concerned that the extension is deliberately left unmaintained after some extension API change, breaking it in OpenLens but continuing to be maintained in Desktop, or a "bug" is introduced that gets patched out of the copy included in Desktop.

**Comment by jakolehm** [comment]
> @jakolehm if you want to share the real reason, e.g. The extensions are closed source now, sources are not longer availble for bussiness reasons. Everyone will understand this and may stop asking for sources.

Removing extensions from this repository has pure technical reasons, we don't want to have any extensions in this repository to make OpenLens more pluggable and extensible (it can be consumed even as a npm package in a near future). Where those past in-tree extensions should live (and be maintained) is a good question which we as a open source community need to figure out. This discussion should probably happen on [Lens Community Slack](https://k8slens.slack.com) `#lens-extensions` channel (to not spam this issue more).

> 
> the whole Community here does not understand the reason, why extensions are removed for technical reasons while there exists does streaam.
> 

Every downstream ("distro") can add functionality through extensions if they wish so. This applies to all OpenLens based distros (Lens Desktop included). Lens Desktop is currently still based on OpenLens 6.2.x series which has these in-tree extensions -> Lens Desktop will have to figure out a way to keep the similar functionality.

> @jakolehm are you accpeting PR to re integrate the extensions?

Not to this repository (see reasons above).

**Comment by hardenchant** [comment]
> @alebcay/openlens-node-pod-menu

according to your, wrote extension installation guide, enjoy :)

https://github.com/alebcay/openlens-node-pod-menu/issues/3#issue-1514506747

**Comment by marafa-sugarcrm** [comment]
47 ppl already installed the extension? nice. that means there are at least 47 ppl using the openlens version

**Comment by alebcay** [comment]
> Lens Desktop is currently still based on OpenLens 6.2.x series which has these in-tree extensions -> Lens Desktop will have to figure out a way to keep the similar functionality.

Ah, didn't realize that. The different versioning between Lens Desktop vs this repo (and the fact that a release of Lens Desktop was dropped on 2022-12-22, right after 6.3.0 got tagged) does not make that clear.

**Comment by vaidik** [comment]
> My team and I, a pretty large global team, have been evaluating Lens for the last 3 months and it was looking really promising. This change has made our decision for us. We will look elsewhere now.

You should checkout [Devtron](https://github.com/devtron-labs/devtron) which is also open-source and covers most features of Lens, and more.

**Comment by EugenMayer** [comment]
With this change, i found myself looking for alternatives too. Interestingly it seems like [k9s](https://k9scli.io/) does basically all of the things i used to have with lens. It's vim like, one needs some adoption time since there is no menu. Just start with `pulses` and then use the numbers 1...8 to navigate as a starter.

OpenLens does work too with the extension installed, but it's future, as for many, does seem uncertain and that is my current backup plan (or will even become plan a))

I do not want to be hostile or promote anything. I'am just a regular user of Lens, then OpenLens and now had to looks for something else and wanted to share my findings.

**Comment by shishirkh** [comment]
This can become a serious problem given that a lot of DevOps folks are used to Lens. Me included! Though I loved using Lens - the new changes have reduced its utility. For now, I have explored two or three tools that can be used instead of Lens.  I tried out K9s, Devtron, and Skooner as replacements.

Out of these, K9s is the only one that was CLI based so it does not give you Lens-like GUI but can still cover basic use cases. Devtron has a nice UI and comes close to what Lens can do. Its free features include the ones that Lens has unfortunately made premium. It's a good bet. Skooner also has a GUI but I don't think it offers any feature that lets you exec into pods without writing kubectl commands. To my surprise, Skooner offers mobile view i.e. you can check k8s cluster things through mobile. Pretty cool feature!

For now, I think Devtron is a good bet. It's open source too! I hope Lens rolls back this change and just never removes imp. features like this!

**Comment by lukjaw** [comment]
I also tested some others like Octant, Kubernetic, InfraApp or KubeWise but none of them have such vast functionality as Lens and almost all are subscription based model too..

**Comment by AssafKatz3** [comment]
> > @jakolehm if you want to share the real reason, e.g. The extensions are closed source now, sources are not longer availble for bussiness reasons. Everyone will understand this and may stop asking for sources.
> 
> Removing extensions from this repository has pure technical reasons, we don't want to have any extensions in this repository to make OpenLens more pluggable and extensible (it can be consumed even as a npm package in a near future). Where those past in-tree extensions should live (and be maintained) is a good question which we as a open source community need to figure out. This discussion should probably happen on [Lens Community Slack](https://k8slens.slack.com) `#lens-extensions` channel (to not spam this issue more).
> 
> > the whole Community here does not understand the reason, why extensions are removed for technical reasons while there exists does streaam.
> 
> Every downstream ("distro") can add functionality through extensions if they wish so. This applies to all OpenLens based distros (Lens Desktop included). Lens Desktop is currently still based on OpenLens 6.2.x series which has these in-tree extensions -> Lens Desktop will have to figure out a way to keep the similar functionality.
> 
> > @jakolehm are you accpeting PR to re integrate the extensions?
> 
> Not to this repository (see reasons above).

This would be a great idea except you blocked also the option to join it, the link from Lens site points to [useless URL](https://join.slack.com/t/k8slens/shared_invite/zt-1cw4vfjkx-962UYNgdQILW7qdbp2GVhA).

**Comment by jakolehm** [comment]
> This would be a great idea except you blocked also the option to join it, the link from Lens site points to [useless URL](https://join.slack.com/t/k8slens/shared_invite/zt-1cw4vfjkx-962UYNgdQILW7qdbp2GVhA).

Slack introduced this limitation while back (link expires after X users join) and we have to constantly regenerate those links. Sorry for the inconvenience (we are trying to find better solution for the Slack invite link).

**Comment by AssafKatz3** [comment]
Hi,
Found the [discussion](https://k8slens.slack.com/archives/C010V2DFMK7/p1671714315155939) and replied to it.

**Comment by SergioSuarezGil** [comment]
bye lens!

**Comment by rizlas** [comment]
Oh this again! Just archive this repo and vanish pls

**Comment by miskun** [comment]
I feel it's time to conclude this discussion as I feel people commenting here don't have a full picture of the situation we are dealing with. To recap:

- We are making OpenLens better.
- We are now breaking down the massive monolith into smaller, more manageable & maintainable components. As an outcome we'll get fast application startup times, less failures on boot, better control over local FS permission issues, ability to address various security issues AND greatly simplified build process…
- At first pass, we ejected all in-tree extensions. **IMPORTANT NOTE:** We'll be ejecting more stuff from the core in the future. This is for the benefit of OpenLens and anybody who relies on Lens for their daily work.
- While ejecting in-tree extensions, we didn't (yet) decide where these ejected components would live. We had various ideas but it was no top priority. As you can imagine, these ejected components are also used by the official Lens Desktop application available from https://k8slens.dev. We'd need to resolve this topic before we can even ship a new version ourselves!
- We have now decided to go ahead with monorepo architecture. The stuff that we will eject from OpenLens "core" will be found in the future from this same repo but they are not included in the "core". They are independent NPM packages that may be loaded into "core". You can follow the progress in here: https://github.com/lensapp/lens/issues/6873

While I understand all the frustration and confusion around this topic, I think it's been getting a bit out of hand. It's quite funny for me to think why some people have a tendency to think that we (people who created Lens to become what it is today) are somehow on a mission to sabotage this project on purpose. Let me ask you: how many projects with >20k stars on Github have you ever created? Let's imagine you have created one. Why would you then start sabotaging it? I guess nobody would. It’s something to be proud of and you’ll want to take super good care of it!

We are quite proud of what we have created and want to keep on working on this project to make it even more awesome in the future. Something that works for our own needs but also for those who want to build their own versions and flavors.

I hope this brings clarity for the issue. We appreciate all constructive feedback on this matter but if the comments in here are just expressing your frustration or suggestions for using X, Y or Z instead of using Lens we’ll close the commenting as it does not have value for the issue.

**Comment by alebcay** [comment]
>  I hope this brings clarity for the issue.

Firstly, I appreciate the work that's being done with the restructuring. I'd like to posit that at least part of the confusion (for myself at least, and I suspect at least for some others) comes from a misunderstanding (or misguided assumption) about the relationship between the open source organization https://github.com/lensapp and Mirantis. Specifically, how much influence does Mirantis product management and corporate priorities have on the direction and governance of this organization and projects contained within it?

> Why would you then start sabotaging it? I guess nobody would. It’s something to be proud of and you’ll want to take super good care of it!

Based on the above misunderstanding, the answer, quite grimly, is "I wouldn't put it past Mirantis". Considering that [`docker-machine` (6.5k stars)](https://github.com/docker/machine) died after Mirantis enacted its new EULA for Docker Desktop, I'm not surprised that people immediately jumped to the worst case conclusion that OpenLens was about to be Mirantis' latest "extend and extinguish" victim.

To be clear, I don't mean to suggest that any maintainer here would rationally want to sabotage this project; rather, it just seems possible that some pressure came down from above (product management, executive decisions, etc.) to deliberately cripple the open-source offering in order to make the proprietary version more appealing.

Answers to the questions I posed above would help to allay such fears and make such an event seem less possible.

> We appreciate all constructive feedback on this matter

I think there are a few things that could be improved in retrospect:

- **Establish clear boundaries and responsibilities with respect to governance and clearly disclose all interests, including commercial ones.** It's not very clear who owns this project and the direction that it goes - the open source community or Mirantis, or, more importantly, who gets the final say when there's a disagreement. If these roles and boundaries are established ahead of time, it's harder to assume malice on the part of project maintainers or Mirantis itself. _I think this is the most important item in this list of suggestions I have here_ - collaboration dies when a community is no longer able to assume good faith about each other (contributors and maintainers alike).
- **Announce larger project direction such as restructuring plans ahead of time.** For many folks (including myself), the removal of the extensions came out of the blue. There's some clues about the restructuring effort that's happening behind the scenes based on reading commit logs, open PRs, etc. but it's not very clear exactly what the end goal is (a better organized and more maintainable project). This lack of clarity leads to speculation and panic about the direction of the project (especially when not assuming good faith, see my previous point). Consider opening story or epic issues well ahead of any breaking changes and pin them at the top of the list so that people can see what's coming up and what to expect.
- **Ensure feature parity in new releases, or loudly announce when features are lost.** Personally, I would suggest that a new home for existing in-tree extensions should have been established and source for those extensions made available before tagging release 6.3.0. Yes, someone could just dig them out of 6.2.6 but making the source for those extensions available would be perceived as an act of confidence/goodwill that you really are just moving the extensions out/around, not trying to commercialize the feature.
- **Mind the optics of when you make disruptive changes and drop releases.** The timing of the 6.3.0 release came at an inopportune time. Keep in mind that many people moved from Lens Desktop to OpenLens to escape the EULA changes that Mirantis imposed on Lens Desktop - this userbase is probably already not very pleased with Mirantis. Consider that the new Mirantis EULA takes effect in the beginning of 2023, and that the OpenLens repo drops a release at the end of 2022 that loses a crucial feature. It's not difficult to jump to the conclusion that the whole thing is a ploy to pressure OpenLens "freeloaders" to use Lens Desktop instead (again, see first point about being able to assume good faith).

**Comment by disconn3ct** [comment]
> Why would you then start sabotaging it?

Because when it was small, it was put under the MIT license and that isn't allowing "me" to take it in a direction that pays bills. And now "I" am trying to fix that without losing that community. Unfortunately, where Mirantis and "I" differ is around conversion of those free users. You are repeatedly alienating that super valuable 20k number, and expecting that we won't all have a bad taste when it comes time to spend money..

My example above makes the same, hugely invalid assumption as the quote. It strongly implies that Mirantis is not involved/in charge, and this is just a misunderstanding amongst open source developers. That is absolutely not the case, and has not been since they showed up and began Mirantising it..

From the shell plugin readme (emphasis mine):

> Like the OpenLens repository itself **at the point from which this extension is based** upon, the content of this repository is released under the MIT license

When will the license of this repo be changing?

**Comment by slok-foundry** [comment]
Guys, I found this plugin much useful for pod logs and terminal.
Add below plugin name in extensions tab in OpenLens
**@alebcay/openlens-node-pod-menu**

**Comment by inquinity** [comment]
At the risk of beating a proverbial dead horse, I have to agree that this was a badly timed and communicated change. I represent several hundreds of users at a fortune 10 company, and we are in the process of switch from Lens to OpenLens. I tested OpenLens 6.2.5, found that it works acceptably, and recommended the upgrade path. I have been receiving numerous support requests from people who went straight to version 6.3.0 and were understandably frustrated with the missing functionality. Sadly, I think this has damaged OpenLens' reputation and lost users.

I understand the intention, but the lack of communication and visible roadmap created huge confusion at a time of maximum impact (due to the volume of new OpenLens users).

For my part, I will keep my development teams on 6.2.5 until the functionality is restored; however, that has created a great deal of extra work just communicating with my internal users.

**Comment by javier-tarazaga** [comment]
@inquinity buy you can get that functionality back in the latest version. In 6.3.0 you can simply open the `Extensions` page, type `@alebcay/openlens-node-pod-menu`, click install and voila, logs and shell are back again into OpenLens. 

I can understand the confusion around the miscommunication and the fact that those features disappeared from the product with an update and was not even notified in the app itself that it was the case (I had to search online myself to figure out what in the world happened), but I guess by simply letting the OpenLens team simply communicate were you can get back the functionality lost before the update by installing extensions which were installed by default I guess would simply solve the issue. 

For OpenLens team, maybe a slight suggestion is that in the Extension page you have a list of known extensions and possibly mark some of them as "oficial" or "reviewed" or similar), with some basic info and a link to the source, add a possibility to simply click `install` and good to go. This would heavily reduce the friction of finding new extensions.

**Comment by halllo** [comment]
@javier-tarazaga Thanks! I started to get familiar with `kubectl logs <pod name>` again :)

**Comment by paul-civitas** [comment]
> * While ejecting in-tree extensions, we didn't (yet) decide where these ejected components would live. We had various ideas but it was no top priority. As you can imagine, these ejected components are also used by the official Lens Desktop application available from https://k8slens.dev. We'd need to resolve this topic before we can even ship a new version ourselves!

@miskun These ejected components are responsible for core functionality we use every day. Have you decided yet whether these ejected components will live in an open source licensed public repo? Or if they will live in a private repo?

The reason people are concerned is because rather than being moved to a different place, such as a different folder or a different open source repo, the components disappeared entirely and a new version was released without any ability to add the components back.

Obviously this functionality isn't going to be removed from the commercial version of lens, so what happened has people thinking that these ejected components are going to be living in a private git repository from now on. You could calm tensions a lot by promising that these components will be open sourced.

**Comment by paul-civitas** [comment]
> We had various ideas but it was no top priority. 

> We'd need to resolve this topic before we can even ship a new version ourselves!

This is strange phrasing. If it's a blocker for a shipping a new version why isn't it a top priority?

**Comment by dj-thd2** [comment]
Thank you for ruining a wonderful opensource project, I will not use Lens nor recommend it anymore to anyone else

**Comment by omniproc** [comment]
Classic RedHat move.

**Comment by inquinity** [comment]
> @inquinity buy you can get that functionality back in the latest version. In 6.3.0 you can simply open the `Extensions` page, type `@alebcay/openlens-node-pod-menu`, click install and voila, logs and shell are back again into OpenLens.
> 
> I can understand the confusion around the miscommunication and the fact that those features disappeared from the product with an update and was not even notified in the app itself that it was the case (I had to search online myself to figure out what in the world happened), but I guess by simply letting the OpenLens team simply communicate were you can get back the functionality lost before the update by installing extensions which were installed by default I guess would simply solve the issue.
> 
> For OpenLens team, maybe a slight suggestion is that in the Extension page you have a list of known extensions and possibly mark some of them as "oficial" or "reviewed" or similar), with some basic info and a link to the source, add a possibility to simply click `install` and good to go. This would heavily reduce the friction of finding new extensions.

@javier-tarazaga I appreciate the advice. With up to several thousand developers all looking to me as the "expert" on using OpenLens, having to instruct them all on adding extensions to get essential functionality could be unfeasible.

My point to the development team is that producing a version that installs without the expected and necessary functionality is not workable. My point to the community is that we are al human, give the development team some time to come up with a solution. There are no villains; at the same time, forthright and honest communication is the foundation of the open source community.

**Comment by cx-dbr** [comment]
😒

**Comment by cubed-it** [comment]
😒

**Comment by cubed-it** [comment]
> While I understand all the frustration and confusion around this topic, I think it's been getting a bit out of hand. It's quite funny for me to think why some people have a tendency to think that we (people who created Lens to become what it is today) are somehow on a mission to sabotage this project on purpose. Let me ask you: how many projects with >20k stars on Github have you ever created? Let's imagine you have created one. Why would you then start sabotaging it? I guess nobody would. It’s something to be proud of and you’ll want to take super good care of it!
> 
> We are quite proud of what we have created and want to keep on working on this project to make it even more awesome in the future. Something that works for our own needs but also for those who want to build their own versions and flavors.
> 
> I hope this brings clarity for the issue. We appreciate all constructive feedback on this matter but if the comments in here are just expressing your frustration or suggestions for using X, Y or Z instead of using Lens we’ll close the commenting as it does not have value for the issue.

Then why is this PR still open and never got feedback?
https://github.com/lensapp/lens/pull/5772

**Comment by jamiecarl** [comment]
At a minimum the extension should be listed in the Lens Extensions repository which is linked from the app extensions page.  To that affect, if I have created a PR with the minimum required changes.  See: https://github.com/lensapp/lens-extensions/pull/21

As a new user of OpenLens, it was MUCH harder to find this functionality than it should have been.  I kinda understand why it was removed and why extensions shouldn't be in the main repo, but honestly this functionality should not be implemented as an extension in the first place.  It is the reason I use this app.

**Comment by emalihin** [comment]
RIP Lens Project. Unless community add-ons save it?

**Comment by ricardo-overhaul** [comment]
from https://github.com/MuhammedKalkan/OpenLens/issues/127#issuecomment-1451852949

> This is due to a breaking change in the Lens project, to move non core functionality into extensions: [lensapp/lens#6823](https://github.com/lensapp/lens/issues/6823) and [lensapp/lens#6983](https://github.com/lensapp/lens/issues/6983)
> 
> Why pods and logs are considered optional? I've got no idea.
> 
> The fix is to install https://github.com/alebcay/openlens-node-pod-menu through the extensions.

**Comment by UmarFKhawaja** [comment]
There is another piece of functionality that has gone missing.

One used to be able to install Prometheus monitoring for the cluster via the Preferences. That has gone missing. Does someone know whether that functionality has also gone AWOL due to project restructuring, and if it is available as an extension?

**Comment by SalahAdDin** [comment]
> from [MuhammedKalkan/OpenLens#127 (comment)](https://github.com/MuhammedKalkan/OpenLens/issues/127#issuecomment-1451852949)
> 
> > This is due to a breaking change in the Lens project, to move non core functionality into extensions: [lensapp/lens#6823](https://github.com/lensapp/lens/issues/6823) and [lensapp/lens#6983](https://github.com/lensapp/lens/issues/6983)
> > Why pods and logs are considered optional? I've got no idea.
> > The fix is to install https://github.com/alebcay/openlens-node-pod-menu through the extensions.

Cool, thanks!

**Comment by madduci** [comment]
> from [MuhammedKalkan/OpenLens#127 (comment)](https://github.com/MuhammedKalkan/OpenLens/issues/127#issuecomment-1451852949)
> 
> > This is due to a breaking change in the Lens project, to move non core functionality into extensions: [lensapp/lens#6823](https://github.com/lensapp/lens/issues/6823) and [lensapp/lens#6983](https://github.com/lensapp/lens/issues/6983)
> > Why pods and logs are considered optional? I've got no idea.
> > The fix is to install https://github.com/alebcay/openlens-node-pod-menu through the extensions.

I've tested the plugin on Windows on latest Open Lens and it still doesn't show the logs

**Comment by andreivasy** [comment]
I've tested https://github.com/alebcay/openlens-node-pod-menu on OpenLens 6.4.11 and it works

**Comment by jamiecarl** [comment]
I created a pull request on this ages ago that still has not been merged.  

https://github.com/lensapp/lens-extensions/pull/21

**Comment by madduci** [comment]
> I've tested https://github.com/alebcay/openlens-node-pod-menu on OpenLens 6.4.11 and it works

I've tested it too with 6.4.11 installed via chocolatey and it doesn't show up

**Comment by SalahAdDin** [comment]
> I created a pull request on this ages ago that still has not been merged.
> 
> [lensapp/lens-extensions#21](https://github.com/lensapp/lens-extensions/pull/21)

I did find the extension, but I don't remember where, hahaha

**Comment by andreivasy** [comment]
> > I've tested https://github.com/alebcay/openlens-node-pod-menu on OpenLens 6.4.11 and it works
> 
> I've tested it too with 6.4.11 installed via chocolatey and it doesn't show up

Is it enabled?
![image](https://user-images.githubusercontent.com/20888658/229158462-9a34d09f-8ab9-4589-a3c5-bc318038d135.png)

I installed OpenLens using the EXE from here: https://github.com/MuhammedKalkan/OpenLens/releases

**Comment by tibistibi** [comment]
Yes. Go  to extensions (ctrl+shift+e) add this name in the box: `@alebcay/openlens-node-pod-menu` and hit install. After that all was a bit buggy but after restart it works fine and i have the menu options back

**Comment by grimlokason** [comment]
it's enabled on 6.4.15, but didn't work :/
going back to the one previous 6.3, 

Quite a bad move to remove fonction like those whle wanting to create a modular apps without providing them.

**Comment by naydoo** [comment]
> Yes. Go to extensions (ctrl+shift+e) add this name in the box: `@alebcay/openlens-node-pod-menu` and hit install. After that all was a bit buggy but after restart it works fine and i have the menu options back

This worked for me

**Comment by alvarezdaniel** [comment]
> Yes. Go to extensions (ctrl+shift+e) add this name in the box: `@alebcay/openlens-node-pod-menu` and hit install. After that all was a bit buggy but after restart it works fine and i have the menu options back

Thanks, shell and logs are back!

**Comment by ChrisNoSim** [comment]
Thx all is working again!!! Nice job 🙏

**Comment by ns-rloffelmacher** [comment]
FWIW I needed to enable the plugin after adding

**Comment by grimlokason** [comment]
The plugin didn't worked here.

tried a lot of things in various order: uninstall/reinstall the plugin. enable/disable it, restart openlens... 
the only solution was to revert to the last version with it.

**Comment by fabio-andre-rodrigues** [comment]
> The plugin didn't worked here.
> 
> tried a lot of things in various order: uninstall/reinstall the plugin. enable/disable it, restart openlens... the only solution was to revert to the last version with it.

The same. After last update no more logs

**Comment by SalahAdDin** [comment]
> > The plugin didn't work here.
> > tried a lot of things in various orders: uninstall/reinstall the plugin. enable/disable it, restart open lens... the only solution was to revert to the last version with it.
> 
> The same. After the last update no more logs

Don't tell us this!

**Comment by kbocock-krg** [comment]
Not working in this version. Literally the logs are the 1 feature I use the most. Please fix asap. OpenLens: 6.5.2-309

**Comment by maximkosheev** [comment]
plugin not work on version 6.5.2-356 even after enabled.

**Comment by SalahAdDin** [comment]
Seems the plugin needs to be updated.

**Comment by FyiurAmron** [comment]
FWIW, please take the discussions about the plugin to the plugin page. It doesn't make any sense to discuss it here. (and it works for me, on OpenLens 6.5.2-356 , you just need to terminate OpenLens right after the install and then restart it, it works then)

**Comment by alebcay** [comment]
For what it's worth, I have been using version 0.1.0 of the plugin with OpenLens 6.5.2-366 without any issues. I'll cut a new release soon but it's unclear to me if this will actually fix any of the problems that people have been running into.

In the future, please use the issue board on the plugin itself for discussing these concerns. I don't actively follow this thread much anymore, so I will be better able to respond to any compatibility issues that people are running into and we can keep track of the outstanding issues in one place.

**Comment by marafa-sugarcrm** [comment]
this was automagically fixed for me

**Comment by gudge25** [comment]
@alebcay/openlens-node-pod-menu
do the trick but  better it to be build in
+ cant find Prometheus installation anymore

**Comment by pbv7** [comment]
> @alebcay/openlens-node-pod-menu do the trick but better it to be build in
> 
> * cant find Prometheus installation anymore

Workaround @shilazi/openlens-lens-metrics

**Comment by swagfin** [comment]
> Yes. Go to extensions (ctrl+shift+e) add this name in the box: `@alebcay/openlens-node-pod-menu` and hit install. After that all was a bit buggy but after restart it works fine and i have the menu options back

This has worked on v6.5.2, thank you. I had to open extension again and manually enable it

**Comment by ro0NL** [comment]
lol why isnt this enabled by default?

**Comment by danilogco** [comment]
> lol why isnt this enabled by default?

They have a paid version and free subscription is gone. Strategy?  :moneybag: 

Btw, if anyone has a problem trying to install extensions, I suggest trying deleting everything from the directory `~/.var/app/dev.k8slens.OpenLens/'. It worked for me. The flatpak version didn't worked well here, I installed via https://github.com/MuhammedKalkan/OpenLens.

**Comment by Laser42** [comment]
> Yes. Go to extensions (ctrl+shift+e) add this name in the box: `@alebcay/openlens-node-pod-menu` and hit install. After that all was a bit buggy but after restart it works fine and i have the menu options back

This doesn't work for me. I tried to force use a registry from my global .npmrc. Was it security restriction or not, i've found the simplest way, because compiling extension from the sources is much more harder.

So:

1. `npm install -g @alebcay/openlens-node-pod-menu`
2. Locate downloaded package, navigate into `@alebcay` folder
3. Use archive tool (I used 7zip) to add `openlens-node-pod-menu` directory into new tar archive
4. Just drag-and-drop tar archive into OpenLens > File > Extensions window

Voila, I hope this can help someone

**Comment by krafcik02** [comment]
In OpenLens open Extension (Win – Ctrl + Shift + e, 🍎 - Command + Shift + e), 
Enter @alebcay/openlens-node-pod-menu  and install
Enjoy!

**Comment by jamiecarl** [comment]
> In OpenLens open Extension (Win – Ctrl + Shift + e, 🍎 - Command + Shift + e), Enter @alebcay/openlens-node-pod-menu and install Enjoy!

Dude, this was closed like a year ago.

**Comment by krafcik02** [comment]
Yesterday I ran into this issue, i have found a solution and so I wanted to
share that. What is your problem?

št 7. 3. 2024 o 23:35 Jamie Carl ***@***.***> napísal(a):

**Comment by kl298p** [comment]
If you migrate from an old machine to a new machine, and want to keep the OpenLens extensions working, it helps quit OpenLens, zip up the ~/.k8slens directory (I'm on Mac) entirely, copy it to the new machine and extract to ~/.k8slens. Only after doing that did the extensions start working for me. I'm on OpenLens 6.5.2-366.

**Comment by asitha-w** [comment]
@alebcay/openlens-node-pod-menu  !!

**Comment by marafa-sugarcrm** [comment]
i use this now https://github.com/freelensapp/freelens  
- 100% open source
- speedier updates
